libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.cpp

```diff
@@ -87,6 +87,11 @@ void AP_BattMonitor_UAVCAN::handle_battery_info(const BattInfoCb &cb)
     _interim_state.temperature = cb.msg->temperature;
     _interim_state.voltage = cb.msg->voltage;
     _interim_state.current_amps = cb.msg->current;
+    _soc = cb.msg->state_of_charge_pct;
+
+    if (_soc > 0 && _soc <= 100) {
+        _use_soc = true;
```

Comment:
This Boolean doesn't seem necessary. Can't you do the same check in the capacity accessor function?

Reply:
When state_of_charge_pct is at [1,100], this boolean is true, indicating that state_of_charge_pct is valid. Until it is true, when state_of_charge_pct is always 0, this boolean is false, indicating that state_of_charge_pct is invalid, use AP_BattMonitor_Backend :: Capacity_remaining_pct(). @magicrub

Reply:
@magicrub @tridge Is there a problem?

Reply:
@magicrub  I think we do need the persistent boolean - we may give the user readings from 100 down t0 1 - if the device reports 0 we want to pass that on, not switch to the parent class function.

Reply:
by the spec, zero is a valid value